docs: added collaboration docs (#631)
SwissDataScienceCenter/renku: docs/user/collaboration.rst

.. _collaborating:

Collaborating on Renku
======================

Beyond building and maintaining your own data science workflows, there are many
ways to use Renku to share your project with collaborators.

Finding & accessing public projects
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To find and access projects on the Renku platform that you might want to
interact with, you can click the **Projects** link in the menu bar followed by
the **Search** tab. Without entering any search terms, you will see a list of

> all public projects. You can click on the project names to be taken to the

Reviewer: ```suggestion
all projects you have access to. You can click on the project names to go to the
```

Author: hmm -- I think I was making a distinction between "public"  = "everyone can read", and "access" = "you can run these notebooks". I should check this everywhere and add a third term.

Reviewer: the search shows you all the projects you can read - it might mean you have guest access, so you still won't be able to launch notebooks, but you can see the project and presumably fork it.

project's landing page.

.. warning::

  Note that the search box currently only allows you to search by project title,
  not by username.

By default, the files in public projects created by other people are
"read only" to you. This means you can see an overview, create and read Kus,
browse the filesystem in both text and :ref:`lineage` modes. The
project's **Notebook Servers** tab will inform you that you do not have
permissions to launch a notebook or make modifications to the code or datasets.

To get access to the project, you can:

* ask the project creator to `add you as a developer <added_to_project_>`_ or maintainer of the project through GitLab.

If the project owner has not added you to the project, you can still participate:

* through `Renku platform "Ku"s <make_kus_>`_, where you can post notes and discussion.

> * through `forking a project <forks_for_collaboration_>`_ through the Renku platform, which creates your own copy of the current state of the project that you own.

Reviewer: ```suggestion
* by `forking a project <forks_for_collaboration_>`_, which creates your own copy of the current state of the project.
```

.. _added_to_project:

Add or be added as a developer on a project
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To allow someone to make changes to your project, you can use the GitLab
interface to add them in Settings > Members, with at either Developer or
Maintainer privileges.

Adding them here will make this project appear in their ``Your Projects`` list
on the Renku platform, where the **Notebook Servers** tab for the project will allow
them to launch an interactive environment to work on the project as usual.

.. warning::

>   QUESTION: How are concurrent notebook session edits handled?

Reviewer: what do you mean "concurrent"?

Author: concurrent: two people who have equal access to a project spawn notebooks. what happens when they try to modify files?

Reviewer: their sessions are separate but the repository is shared - so a usual git workflow is recommended, e.g. using branches and merging to avoid conflicts as much as possible

Author: ok, cool! will add that to the docs as well!

.. _make_kus:

Add collaborators to the conversation through "Ku"s
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

In Renku, Kus are media-rich discussions you can use to help keep track of
your work and to collaborate with others. To find out more about them and walk
through sample usage, see :ref:`sharing_is_caring`.

.. _forks_for_collaboration:

> Fork a project take it in your own direction

Reviewer: ```suggestion
Fork a project and take it in your own direction
```

of the project through GitLab.

.. _forks_for_collaboration:

Fork a project and take it in your own direction
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Forking someone else's project on Renku is just like forking any version-controlled
project. The Renku platform provides you a fork button on the project's home page
that creates a fork of the project (with the same name) under your own namespace,
where you have permissions to launch notebooks and make settings modifications.

One reason you might want to fork someone's project is to use it as a template for
your own analysis.

.. warning::

  Note that you will be unable to fork a project that has an identical name to a
  project already in **Your Projects** list. We suggest you change the name of
  your own project (which you can do in the Renku interface) before forking.

.. _make_kus:

Add collaborators to the conversation through "Ku"s
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

In Renku, Kus are media-rich discussions you can use to help keep track of
your work and to collaborate with others. To find out more about them and walk
through sample usage, see :ref:`sharing_is_caring`.

.. _added_to_project:

> Add or be added as a developer on a project

Reviewer 2: A bit late to the party but... found a typo here

Author: oh no! where?! I don't see it!

Reviewer 2: oh sorry, I accidentally modified the code in the local repo, but here everything seems fine.
My bad :man_facepalming:

Forking someone else's project on Renku is just like forking any version-controlled
project. The Renku platform provides you a fork button on the project's home page
that creates a fork of the project (with the same name) under your own namespace,
where you have permissions to launch notebooks and make settings modifications.

One reason you might want to fork someone's project is to use it as a template for
your own analysis.

.. warning::

  Note that you will be unable to fork a project that has an identical name to a
  project already in **Your Projects** list. We suggest you change the name of
  your own project (which you can do in the Renku interface) before forking.

.. _make_kus:

Add collaborators to the conversation through "Ku"s
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

In Renku, Kus are media-rich discussions you can use to help keep track of
your work and to collaborate with others. To find out more about them and walk
through sample usage, see :ref:`sharing_is_caring`.

.. _added_to_project:

Add or be added as a developer on a project
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To allow someone to make changes to your project, you can use the GitLab

> interface to add them in Settings > Members, with at either Developer or

Reviewer: ```suggestion
interface to add them in ``Settings > Members``, with at either Developer or
```

To find and access projects on the Renku platform that you might want to
interact with, you can click the **Projects** link in the menu bar followed by
the **Search** tab. Without entering any search terms, you will see a list of
all projects that you have access to. You can click on a project name to be
taken to the project's landing page.

.. warning::

  Note that the search box currently only allows you to search by project title,
  not by username.

By default, the files in projects created by other people are "read only" to you.
This means you can see an overview, create and read Kus, and browse the filesystem
in both text and :ref:`lineage` modes. The project's **Notebook Servers** tab
will inform you that you do not have permissions to launch a notebook or make
modifications to the code or datasets.

If you would like to interact with the notebooks (run, edit, upload data, continue

> your own analyses, etc.), you can:

Reviewer: probably don't need a list here since there is only one option

* `fork the project <forks_for_collaboration_>`_, which creates your own copy of the current state of the repo.

If you have questions, notes, or discussion topics that you would communicate to

> the project's owner(s), you can:

Reviewer: probably don't need a list here since there is only one option

* Make `Renku platform "Ku"s <make_kus_>`_

> If you are part of the team and want to work on this project, you can:

Reviewer: probably don't need a list here since there is only one option
